icu4c/source/i18n/messageformat2_function_registry.cpp

```diff
@@ -905,6 +906,181 @@ Formatter* StandardFunctions::DateTimeFactory::createFormatter(const Locale& loc
     return result;
 }
 
+static UDate tryPattern(const char* pat, const UnicodeString& sourceStr, UErrorCode& errorCode) {
+    LocalPointer<DateFormat> dateParser(new SimpleDateFormat(UnicodeString(pat), errorCode));
+    if (U_FAILURE(errorCode)) {
+        return 0;
+    }
+    return dateParser->parse(sourceStr, errorCode);
+}
+
+static UDate tryPatterns(const UnicodeString& sourceStr, UErrorCode& errorCode) {
+    if (U_FAILURE(errorCode)) {
+        return 0;
+    }
+    UDate d = tryPattern("YYYY-MM-dd'T'HH:mm:ss", sourceStr, errorCode);
```

Comment:
instead of trying first one then the next one, could you make decision of which one to call by the length of the sourceStr? Isn't that true for the tryPattern("YYYY-MM-dd'T'HH:mm:ss" ) to return error the length of sourceStr must be in a certain length ? same question with the "YYYY-MM-dd".

Reply:
also, why are we trying "YYYY-MM-dd'T'HH:mm:sszzzz" ? and "YYYY-MM-dd"? which spec are you based on to try these two patterns? (and these two patterns only?)

Reply:
Good idea, done!